p2p/connection.go

```diff
@@ -444,8 +458,9 @@ FOR_LOOP:
 			c.Logger.Debug("Receive Ping")
 			c.pong <- struct{}{}
 		case packetTypePong:
-			// do nothing
 			c.Logger.Debug("Receive Pong")
+			// Should we unset pongTimer if we get other packet?
```

Comment:
Good question. Since the correct behaviour is to receive a "pong" in response to a "ping", we should probably insist it is received. Otherwise we could imagine malicious peers that only send us particular kinds of messages, and never respond to our pings, so we keep the conn alive, even though we shouldn't. Forcing the pong probably doesn't buy us much on that front, since it's easy for a malicious peer to implement, but it helps enforce correctness. So I wouldn't unset it here. But maybe @melekes / @petabytestorage / @cloudhead have different opinions

Reply:
Nice reasoning @ebuchman!

Reply:
In WSClient I went with resetting deadline because server might be overloaded and not have a chance to send us pong, but it does not mean it's not correct.